### PR TITLE
go modules support

### PR DESCRIPTION
**Pull request by aidun**
- Go Modules
- Build instructions

Tested wit Apple M1 and Go 1.16

Signed-off-by: Markus Hartmann <markush1986@gmail.com>

### CODE REVIEW

**Comment by aidun** [comment]
Nice project! Love to help you.

**Comment by abrander** [comment]
Thanks for your contribution. It's awesome, gomod support is highly overdue. I have a few thoughts:

- Can we somehow give the `connect` CLI app its own go.mod? The `github.com/abrander/garmin-connect` package has no dependencies in itself, I think it's an unneeded burden on package users if we require packages only needed for the `connect` app for users using the `garmin-connect` package in their own project. Can we have two separate go.mod-files? One in `connect` covering the needs of `connect`, and another in `garmin-connect` covering *that* package and nothing else?

- Shouldn't `go get github.com/abrander/garmin-connect/connect` install the connect app in a properly configured Go environment thereby reducing the install instruction for the CLI app to that one command?

If these issues could somehow be resolved, I would love to merge this and any future PR's :smiley_cat:

**Comment by aidun** [comment]
Thank you for your feedback. I your approach to install the cli and you are right. I changed it.

**Comment by aidun** [comment]
The version with the two go.mod files is working for me. Please test it in your environment and give some feedback. Go Modules are not my sweet spot.

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+connect-cli
```
Comment (abrander) [inline .gitignore:1]: This looks specific to your environment. There's already a .gitignore in `connect`. I would suggest simply removing this file.

File: connect/go.mod [hunk 2]
```diff
@@ -0,0 +1,9 @@
+module github.com/abrander/garmin-connect/connect
+
+go 1.16
```
Comment (abrander) [inline connect/go.mod:3]: I don't think Go 1.16 is needed? At least 1.15 should suffice, probably older too, but I have no (easy) way of testing.

File: go.mod [hunk 3]
```diff
@@ -0,0 +1,3 @@
+module github.com/abrander/garmin-connect
+
+go 1.16
```
Comment (abrander) [inline go.mod:3]: How about we try requiring 1.15, and not 1.16? I believe older versions would suffice too, but I have no easy way of testing it.

**Review by abrander — CHANGES_REQUESTED** [review]
I only have minor comments. Thanks for the `go mod edit -replace` snippet, that always gives me a headache.

Can we lose the empty `go.sum` file?

File: connect/go.mod (hunk 2 above)
Comment (aidun) [reply connect/go.mod:3]: I will change it to 1.15. It is autogenerated by go mod init.

File: .gitignore (hunk 1 above)
Comment (aidun) [reply .gitignore:1]: You are right, not my best PR :-)

**Comment by aidun** [comment]
the sum files will be generated by every go build or go mod tidy ... In some cases the go command requires this file to be present. I would leave it as it is.

File: go.mod (hunk 3 above)
Comment (aidun) [reply go.mod]: I will change it to 1.15. It is autogenerated by go mod init.

**Review by abrander — APPROVED** [review]
This is great! Good call on the empty `go.sum` file. I'll merge right away! Thanks! :smiley_cat:

**Pull request merged**